message_converter/include/converter_node.hpp

```diff
@@ -0,0 +1,200 @@
+#pragma once
+#include "ros/ros.h"
+#include <ros/console.h>
+#include <boost/array.hpp>
+#include <geometry_msgs/PointStamped.h>
+#include <gnss_msgs/BaselinePosition.h>
+#include <gnss_msgs/BaselineVelocity.h>
+#include <gnss_msgs/RtkAvailable.h>
+#include <geometry_msgs/PointStamped.h>
+#include <piksi_rtk_msgs/BaselineNed.h>
+#include <piksi_rtk_msgs/VelNedCov.h>
+#include <libsbp_ros_msgs/MsgBaselineNed.h>
+#include <libsbp_ros_msgs/MsgVelNedCov.h>
+#include <libsbp_ros_msgs/MsgPosLlh.h>
+#include <libsbp_ros_msgs/MsgGpsTime.h>
+#include <eigen3/Eigen/Dense>
+#include <math.h>
+#include <std_msgs/Bool.h>
+
+#include <diagnostic_msgs/DiagnosticArray.h>
+#include <diagnostic_updater/diagnostic_updater.h>
+#include <diagnostic_updater/publisher.h>
+
+#include "libsbp_ros_msgs/ros_conversion.h"
+
+class SwiftNavRover{
+public:
+    SwiftNavRover(ros::NodeHandle* node_handle, double la, double lo, double al);
+
+    // WGS 84 derived geometric constants
+    const double kSemimajorAxis = 6378137.0;
+    const double kSemiminorAxis = 6356752.3142;
+    const double kFirstEccentricitySquared = 6.69437999014 * 0.001;
+    const double kSecondEccentricitySquared = 6.73949674228 * 0.001;
+    const double kFlattening = 1/298.257223563;
+
+    typedef Eigen::Matrix<double,3,3> Matrix3x3d;
+    Matrix3x3d ecef_to_ned_matrix;
+
+    ros::NodeHandle nh;
+    
+    /**
+     * Initialize the publishers and subscribers for the ROS node
+     * @param 
+     * @return 
+     */
+    void init();
+
+    // Publisher functions
+    /**
+     * Publish the baseline position (NED), with mesage type gnss_msgs/BaselinePosition
+     * @param fields of the gnss_msgs/BaselinePosition message type
+     * @return void
+     */
+    void publishBaselinePosition(ros::Time& t, double n, double e, double d, int tow, boost::array<double, 9>& covariance, int n_sats, int fixed_mode);   
+
+    /**
+     * Publish the baseline velocity, with mesage type gnss_msgs/BaselineVelocity
```

Comment:
not blocking but this description is obvious from the method name, the doc comment needs to add additional information like why the method is here. I'd prefer no comment to a redundant one.